src/templates.rs

```diff
@@ -0,0 +1,37 @@
+use log::error;
+use std::fs;
+use std::path::PathBuf;
+
+/// Creates the `templates/` folder and base template file.
+pub fn initialize_templates(output_folder: &PathBuf) {
+    let templates_path = output_folder.join("templates");
+    if let Err(e) = fs::create_dir_all(&templates_path) {
+        error!("Failed to create templates directory: {}", e);
+        return;
+    }
+
+    let base_template_path = templates_path.join("base.html");
+    if let Err(e) = fs::write(base_template_path, "<!-- Base HTML template -->") {
+        error!("Failed to create base template: {}", e);
+    }
+}
+
+/// Creates the `static/` folder with basic placeholder files.
+pub fn initialize_theme_assets(output_folder: &PathBuf) {
+    let static_path = output_folder.join("static");
+    if let Err(e) = fs::create_dir_all(&static_path) {
+        error!("Failed to create static assets directory: {}", e);
+        return;
+    }
+
+    let css_path = static_path.join("style.css");
+    let js_path = static_path.join("script.js");
+
+    if let Err(e) = fs::write(css_path, "/* Add your styles here */") {
+        error!("Failed to create CSS file: {}", e);
+    }
+
+    if let Err(e) = fs::write(js_path, "// Add your scripts here") {
+        error!("Failed to create JavaScript file: {}", e);
+    }
```

Comment:
Here you can call

https://github.com/rochacbruno/marmite/blob/89841683bbc469ce1ac6d65a475ac375ec91b3ef/src/embedded.rs#L45

But passing the `input_folder.join("static")`, if it does not exists yet

The output will be the embedded template, then user can customize before generating the site.

Reply:
The initialize_theme_assets function creates a static folder in input_folder (if it doesn’t already exist) and then calls generate_static(&static_path). This utilizes the generate_static function to populate static with embedded assets, eliminating the need for placeholder files